smolyay/basis.py

```diff
@@ -57,6 +61,23 @@ def derivative(self, x):
         """
         pass
 
+    def outside_domain(self, x):
```

Comment:
Methods should be actions, and I think this is clearer:
```suggestion
    def check_in_domain(self, x):
```
Also, is this part of the public interface to this class? If not, prepend a `_` to indicate it is private.